WPFSKillTree/Utils/UrlProcessing/BuildConverter.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+
+namespace POESKillTree.Utils.UrlProcessing
+{
+    /// <summary>
+    /// Creates an instances of a class derived from the <see cref="BuildUrlDeserializer"/> class.
+    /// </summary>
+    public static class BuildConverter
+    {
+        private static ISet<TryCreateDeserializer> _deserializersFactories = new HashSet<TryCreateDeserializer>();
+        private static Type _defaultDeserializerType;
+
+        /// <summary>
+        /// Registers default deserializer that should be used if all others refused to parse.
+        /// It may be useful when trying to parse obsolete build urls and official site compatible urls.
+        /// </summary>
+        /// <param name="defaultDeserializer">The default deserializer type.</param>
+        public static void RegisterDefaultDeserializer(Type defaultDeserializer)
```

Comment:
It looks nice!
I'll use it with constraint like this:
`public static void RegisterDefaultDeserializer<T>(Func<string, T> factory) where T : BuildUrlDeserializer`